app/src/main/res/values/strings.xml

```diff
@@ -85,21 +85,21 @@
     <string name="segment_skipped">Skipped segment</string>
     <string name="category_segments">Segments</string>
     <string name="category_sponsor">Sponsor</string>
-    <string name="category_sponsor_description">Paid promotion, paid referrals and direct advertisements. Not for self-promotion or free genuine shoutouts to causes, creators, websites, and products.</string>
+    <string name="category_sponsor_description">Paid promotion, paid referrals and direct advertisements. Not for self-promotion or free genuine shoutouts to causes, creators, websites, and products: %s</string>
```

Comment:
Is there a better way to do this? Adding %s to all languages seems tedious and inefficient but I wanted a way to indicate what setting was currently selected.

Reply:
Yes, we can subclass `DropDownPreference` and override `getSummary()`. That's probably way better because using this way, each translator needs to update all the strings too.

Reply:
```kotlin
override fun getSummary(): CharSequence {
val currentIndex = entryValues.indexOf(value)
return "${super.getSummary()} • ${entries[currentIndex]}"
}
```

Reply:
Where would this code go?

Reply:
```kotlin
class SponsorBlockDropDownPreference(context: Context, attrs: AttributeSet): DropDownPreference(context, attrs) {
// insert the cod here
}`
```

Reply:
And then just use that class in the preference XML files instead of `DropDownPreference`. I didn't test it though, might be that the getSummary function needs some minor modifications

Reply:
Do you think we could implement a spinner instead? Kind of like how it was before but instead of the slider we have the spinner with the options. Not entirely sure how to implement it (What I'm not sure about is how to get the xml file to read our custom class but I think thats as simple as adding the class to a certain folder).

Reply:
I don't really see the purpose of it, wouldn't a spinner preference be very similar to a dropdown preference. And it doesn't matter which directory we add the preference class to, we just have to reference it in the XML with `com.github.libretube.ui.views.SponsorBlockPrefernece` for example.

Reply:
So currently the setting looks like this:
<img width="505" alt="Screen Shot 2023-06-17 at 2 31 41 PM" src="https://github.com/libre-tube/LibreTube/assets/48372480/939011b3-4f34-42ed-9d37-b334d7eda374">
I think something like this would look cleaner:
<img width="511" alt="Screen Shot 2023-06-17 at 2 33 58 PM" src="https://github.com/libre-tube/LibreTube/assets/48372480/83b0ce1c-8f2f-4346-9f6e-baaf53465b7e">

What do you think would be best?

Reply:
Oh yes, I see! The second approach probably looks way better. I think we can just override `onCreate` in our custom preference class and set a different layout via `setContentView` and view bindings, and then just set the texts of the layout (title, summary, current value) manually.

Reply:
Alright sounds good. I'll get that going

Reply:
So I'm not the best with front end stuff. So from my understanding the way the settings are drawn needs to be changed for our custom preference. Is this done in the custom preference file itself or in the file that processes the xml file?

Reply:
In the class/file that extends `DropDoenPreference`.

Reply:
Do you know of any good resources/ examples on how to modify these functions?

Reply:
https://overflow.lunar.icu/questions/16108609/android-creating-custom-preference

This looks pretty much what we're doing, just that we override `DropDownPreference` instead of `Preference`. The code in the linked issue is in java, but it works basically the same way in Kotlin.